unit/big-int/big-int.cpp

```diff
@@ -0,0 +1,256 @@
+/*******************************************************************\
+
+Module: Unit test for big-int
+
+Author: Daniel Kroening
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <string>
+
+#include <big-int/bigint.hh>
+
+// =====================================================================
+// Printing and reading bignums.
+// =====================================================================
+
+static std::string to_string(BigInt const &x, unsigned base = 10)
+{
+  const std::size_t len = x.digits(base) + 2;
+  std::vector<char> dest(len, 0);
+  const char *s = x.as_string(dest.data(), len, base);
+  return std::string(s);
+}
+
+static bool read(const std::string &input, BigInt &x, unsigned base = 10)
+{
+  return x.scan(input.c_str(), base) == input.c_str() + input.size();
+}
+
+SCENARIO("arbitrary precision integers", "[core][big-int][bigint]")
```

Comment:
Given these aren't really `given/when/then` style tests you might want to just use the `TEST_CASE` header (see https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#test-cases-and-sections)

Reply:
Thanks a lot for the guidance, done as suggested.